api/lib/session/bloc/session_bloc.dart

```diff
@@ -207,6 +208,34 @@ class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
 
   void _onTicked(_TimerTicked event, Emitter<SessionState> emit) {
     emit(state.copyWith(remainingTime: event.duration));
+
+    /// Dont show the answer for first 10 seconds
+    if (event.duration < 51) {
+      final difference = (event.duration.toDouble()) / 10;
+
+      /// If the timer hits 50,40,30,20 s time mark
+      if ((difference is int || difference.toInt() == difference) &&
+          difference <= 5 &&
+          difference > 1) {
+        var hiddenAnswer = state.hiddenAnswer;
+
+        /// choose a random char to show from [correctAnswer]
+        final charIndex = Random().nextInt(state.correctAnswer.length - 1);
+
+        hiddenAnswer = replaceCharAt(
+          state.hiddenAnswer,
+          charIndex,
+          state.correctAnswer[charIndex],
+        );
+
+        emit(
+          state.copyWith(
+            hiddenAnswer: hiddenAnswer,
+          ),
+        );
```

Comment:
How are we ensuring that all letters will be shown when the timers ends? i.e. the logic to show number of letters at every mark according to the word length

Reply:
I didnt understand,
we are showing some letters as a hint to the players
we dont need to show all letters, correct answer will be shown when the round is complete.

Reply:
let's say we have a very long word and the timer is only of 60 seconds, in this case we need to show more than one letter at 50/40/30 mark.

Reply:
Is this critical for our first prototype? we can include this in refinements for now